### PR TITLE
feat: fractional mainnet migration

### PR DESCRIPTION
**Pull request by rndquu**
This PR refactors mainnet and development migrations to reflect changes introduced in https://github.com/ubiquity/ubiquity-dollar/pull/922

The main changes are:
1) Curve's Governance-WETH crypto pool:
  - mainnet: use already deployed contract at https://etherscan.io/address/0xacdc85afcd8b83eb171affcbe29fad204f6ae45c
  - anvil/testnet: deploy a mock from scratch
2) Chainlink price feed for ETH/USD pair:
  - mainnet: use already deployed price feed at https://etherscan.io/address/0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419
  - anvil/testnet: deploy a mock price feed from scratch
3) Small updates like setting collateral ratio to 95% from start, calling some necessary setters, etc...

P.S. All tests are passed for anvil instance forked from mainnet in [this](https://gist.github.com/rndquu/6797b0d7b9110e0b8faae6304ca600d9) test file

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -96,6 +96,20 @@ COLLATERAL_TOKEN_ADDRESS="0x5f98805A4E8be255a32880FDeC7F6728C6568bA0"
 # - testnet/anvil: deploys LUSD/USD chainlink price feed from scratch
 COLLATERAL_TOKEN_CHAINLINK_PRICE_FEED_ADDRESS="0x3D7aE7E594f2f2091Ad8798313450130d0Aba3a0"
 
+# Curve's Governance/WETH pool address.
+# Used to fetch Governance/ETH price from built-in oracle.
+# By default set to already deployed (production ready) Governance/WETH pool address.
+# - mainnet: uses already deployed Governance/ETH pool address
+# - testnet/anvil: deploys Governance/WETH pool from scratch
+CURVE_GOVERNANCE_WETH_POOL_ADDRESS="0xaCDc85AFCD8B83Eb171AFFCbe29FaD204F6ae45C"
```
Comment (gitcoindev) [inline README.md:104]: Address verified on Etherscan and https://www.geckoterminal.com/eth/pools/0xacdc85afcd8b83eb171affcbe29fad204f6ae45c

File: README.md [hunk 2]
```diff
@@ -96,6 +96,20 @@ COLLATERAL_TOKEN_ADDRESS="0x5f98805A4E8be255a32880FDeC7F6728C6568bA0"
...
+
+# Chainlink price feed address for ETH/USD pair.
+# Used to calculate Governance token price in USD.
+# By default set to ETH/USD price feed deployed on ethereum mainnet.
+# - mainnet: uses already deployed ETH/USD chainlink price feed
+# - testnet/anvil: deploys ETH/USD chainlink price feed from scratch
+ETH_USD_CHAINLINK_PRICE_FEED_ADDRESS="0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419"
```
Comment (gitcoindev) [inline README.md:111]: Verified Pair	Deviation	Heartbeat	Dec	Address and info
[🟢](https://docs.chain.link/data-feeds/selecting-data-feeds#-low-market-risk-feeds)ETH / USD
0.5%	3600s	8	
[0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419](https://etherscan.io/address/0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419)
Asset name:
Ethereum

File: packages/contracts/migrations/development/Deploy001_Diamond_Dollar_Governance.s.sol [hunk 3]
```diff
@@ -482,12 +507,141 @@ contract Deploy001_Diamond_Dollar is Script, DiamondTestHelper {
...
+        vm.stopBroadcast();
+
+        //=============================================
+        // Curve's Governance-WETH crypto pool setup
+        //=============================================
+
+        // start sending admin transactions
+        vm.startBroadcast(adminPrivateKey);
+
+        // set ETH/Governance price to 30k in Curve pool mock
```
Comment (gitcoindev) [inline packages/contracts/migrations/development/Deploy001_Diamond_Dollar_Governance.s.sol:635]: Ok, if i got correctly the mocked settings, ETH/USD set to $3000 and ETH/Governance set to 30k, Governance should be ~0.1 USD

File: packages/contracts/migrations/mainnet/Deploy001_Diamond_Dollar_Governance.s.sol [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+     * There are 2 migrations (deployment scripts):
+     * 1. Development (for usage in testnet and local anvil instance)
+     * 2. Mainnet (for production usage in mainnet)
+     *
+     * Mainnet (i.e. production) migration uses already deployed contracts for:
+     * - Chainlink collateral price feed contract
+     * - UbiquityAlgorithmicDollarManager contract
+     * - UbiquityGovernance token contract
+     * - Chainlink ETH/USD price feed
+     * - Curve's Governance-WETH crypto pool
```
Comment (gitcoindev) [inline packages/contracts/migrations/mainnet/Deploy001_Diamond_Dollar_Governance.s.sol:73]: Good summary. Price feed relations are getting complex so best is to keep the explanations close to code.

File: packages/contracts/migrations/mainnet/Deploy001_Diamond_Dollar_Governance.s.sol [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+        // UbiquityGovernance setup
+        //============================
+
+        // NOTICE: If owner address is `ubq.eth` (i.e. ubiquity deployer) it means that we want to perform
+        // a real deployment to mainnet so we start sending transactions via `startBroadcast()`. Otherwise
+        // we're in the forked mainnet anvil instance and the owner is not `ubq.eth` so we can't add "UBQ_MINTER_ROLE"
+        // and "UBQ_BURNER_ROLE" roles to the diamond contract (because only `ubq.eth` address has this permission).
+        // Also we can't use "vm.prank()" since it doesn't update the storage but only simulates a call. That is why
+        // if you're testing on an anvil instance forked from mainnet make sure to add "UBQ_MINTER_ROLE" and "UBQ_BURNER_ROLE"
+        // roles to the diamond contract manually. Take this command for inspiration:
```
Comment (gitcoindev) [inline packages/contracts/migrations/mainnet/Deploy001_Diamond_Dollar_Governance.s.sol:180]: Note: This description is very important, especially for migration testing.

File: packages/contracts/migrations/mainnet/Deploy001_Diamond_Dollar_Governance.s.sol [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+
+        // deploy Curve Dollar-3CRV metapool
+        address curveDollarMetaPoolAddress = ICurveStableSwapFactoryNG(
+            0x6A8cbed756804B16E05E741eDaBd5cB544AE21bf
+        ).deploy_metapool(
+                0xbEbc44782C7dB0a1A60Cb6fe97d0b483032FF1C7, // Curve 3pool (DAI-USDT-USDC) address
+                "Dollar/3CRV", // pool name
+                "Dollar3CRV", // LP token symbol
+                address(dollarToken), // main token
+                100, // amplification coefficient
```
Comment (gitcoindev) [inline packages/contracts/migrations/mainnet/Deploy001_Diamond_Dollar_Governance.s.sol:135]: Note: https://curve.readthedocs.io/factory-deployer.html#choosing-an-amplification-coefficient

**Review by gitcoindev — APPROVED** [review]
Approved from my side, just added a few notes / comments. Good job!

**Review by molecula451 — APPROVED** [review]
Clean & Clear PR

**Pull request merged**